### PR TITLE
Add a Code Rights Declaration

### PR DESCRIPTION
**Pull request by vaxerski**
This adds a code rights declaration, a short DCO/CLA-style document to agree to when submitting a PR.

Everything is up to discussion, obviously.

cc @fufexan 

The link in the PR template is broken atm (because it points to parent) but will work once merged.

### Reasoning

Hyprland is growing in popularity and I want to make sure no big entities try to fuck us over like Apple did with BSD.

GPL is too restrictive.

Basically, I will never make this closed source, but I want the rights to at one point say "commercial usage only under some conditions under certain terms"

This is a very lenient agreement, and I will not pull a MongoDB on y'all, pinky promise.

### Your input

I am open to suggestions, changes, and opposing views. Feel free to voice them.

### CODE REVIEW

**Comment by kojq** [comment]
I am on board.

**Comment by DashieTM** [comment]
As it currently stands, the agreement would ensure source available but not open source.
I get the idea and while I do believe you that you won't take down any forks, it's still not perfect, and a guarantee is better than a promise.

"as long as that license does not hinder the ability of one to contribute to, or view the source code of the project."

Would LGPL be something to consider? Or rather what restriction from the GPL license should be removed for it to be suitable?

**Comment by outfoxxed** [comment]
Basically agree with dashie, LGPL seems like it would be good, do you have any specific cases you want to allow or prevent?

**Comment by vaxerski** [comment]
> As it currently stands, the agreement would ensure source available but not open source.
I get the idea and while I do believe you that you won't take down any forks, it's still not perfect, and a guarantee is better than a promise.

Suggest a change, I don't see how this makes me allowed to take down forks?

File: docs/RIGHTS_DECLARATION.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+ Ⅰ) This contribution is made entirely by myself, or in a way that
+ gives me full rights to the contribution.
+
+ Ⅱ) I realize and understand this contribution will be permanently recorded
+ under this repository alongside the information attached to it (Such as my Github username)
+
+ Ⅲ) I give full rights to Hypr Development to redistribute this contribution as a part of the parent project in both source and binary form.
+
+ Ⅳ) I give full rights to Hypr Development to relicense the parent project to which I am contributing at a later date to any license as long as that license does not hinder the ability of one to contribute to, or view the source code of the project.
```
Comment (DashieTM) [inline docs/RIGHTS_DECLARATION.md:21]:
```suggestion
 Ⅳ) I give full rights to Hypr Development to relicense the parent project to which I am contributing at a later date to any license as long as that license does not hinder the ability of one to contribute to, redistribute or view the source code of the project.
```

Comment (vaxerski) [reply docs/RIGHTS_DECLARATION.md:21]: See I don't want to add that because imagine a scenario one day where a corporation starts using any part of my projects, and I want a cut from them. I have no leverage, as this disallows me from adding a clause prohibiting commercial use / use by commercial entities.

Comment (DashieTM) [reply docs/RIGHTS_DECLARATION.md:21]: In that case you will have to abandon the OSS definition i'm afraid.
For that usecase I would simply add non-commercial in front of it and be done with that.

I am in doubt if that holds up legally though.

Comment (vaxerski) [reply docs/RIGHTS_DECLARATION.md:21]: OSS definition? Where do I define as that? You mean line 9?

Also, I believe it's more like dropping the F rather than the OSS xD

How about `license does not hinder the ability of any non-commercial entity to contribute to, redistribute or view the source code of the project.`

Comment (DashieTM) [reply docs/RIGHTS_DECLARATION.md:21]: I guess that would do it.

About the OSS definition, I based that on the open source initiative, I would have pasted the section about this, but somehow they managed to let their certificate expire...... they pulled a manjaro

Comment (vaxerski) [reply docs/RIGHTS_DECLARATION.md:21]: In all honesty - based on a lot of the open source initiative points I've seen - it's a damn pipe dream and completely unrealistic for the modern world. You will _not_ get rid of the tech giants fucking people over in any way other than by law.

Comment (vaxerski) [reply docs/RIGHTS_DECLARATION.md]: committed, thanks!

**Comment by morr0ne** [comment]
While I do agree that a CLA of sorts is a good idea this is not the proper solution. If the goal is to avoid getting "fucked over" like apple with BSD the current proposal is not enough.
A CLA is for all intense a purpose a binding legal document that should use unambiguous language and be legally enforceable in most jurisdictions. If not it becomes nothing more than a piece of text companies can easily look over. It should also include two separate versions, an individual license agreement and an Entity license agreement that can be signed on behalf of legal entities.

Unless someone wants to hire multiple lawyers and go through the legal system of different countries the most sensible solution would be do adopt the [Fiduciary Licence Agreement (FLA)](https://fsfe.org/activities/fla/fla.en.html) which is made specifically for this purpose.  It is actively adopted and enforced by FOSS projects, most notably KDE. 
A purpose made version can be created [here](https://contributoragreements.org/ca-cla-chooser/)

More information about this stuff can be found on the [Contributor Agreements website](https://contributoragreements.org/)

Finally the CLA signing processed should be automated with tools like https://cla-assistant.io instead of requiring manual user intervention

All of this might seem pedantic but I do believe something like a CLA should be done properly otherwise it just becomes an annoyance for contributors without actually protecting the project in any meaning-full way

**Comment by Sivecano** [comment]
Maybe license the project as lgpl but in the CLA reserve the ability to relicense.
You can release the code under different conditions but any existing forks are untouchable since the license they're granted is perpetual.
This could worst-case lead to another asprite/libresprite.

**Comment by morr0ne** [comment]
> I'll check it out. The biggest problem with "estabilished" CLAs is their pedantic nature that might scare contributors, and often lack of permissiveness towards me, the owner. Though I see an option to customize the FLA in there, this could be nice.

It is fairly customizable. I do agree this documents can be quite scary for certain people but unfortunately there really is no way around it. Big corporations have deep pockets and armies of lawyers, simple documents like the original proposal are just not enough.

**Comment by vaxerski** [comment]
Makes sense. I'll probably take the FLA, generate a close license and tweak a few terms.

I want to keep this fully foss for _individuals_ and _non-profits_, but reserve the rights to give a middle finger to _for-profits_

**Comment by morr0ne** [comment]
The FLA is meant to protect most parties but most importantly it ensures that the projects does not become proprietary

> Makes sense. I'll probably take the FLA, generate a close license and tweak a few terms.

Changes should be done carefully since they might make the document invalid. It's probably a good idea to document the changes and ask for inputs before committing to it

> I want to keep this fully foss for _individuals_ and _non-profits_, but reserve the rights to give a middle finger to _for-profits_

That's mostly down to just changing the license which any good CLA like this one allows. It is important tho that current and past contributors allow it since a CLA for obvious reasons isn't retroactively enforceable.

**Comment by vaxerski** [comment]
That I understand.

On a side note - since I write 95% of hyprland's code, wouldn't it be possible for me to one day add a clause that all _my_ contributions from here on out can't be used by commercial entities...?

**Comment by jaminW55** [comment]
Would it not be good to add something along the lines of "any program which uses this source in all or in part, regardless of commercial status, must make all associated code of said program open source"

**Comment by morr0ne** [comment]
> On a side note - since I write 95% of Hyprland's code, wouldn't it be possible for me to one day add a clause that all _my_ contributions from here on out can't be used by commercial entities...?

Technically speaking you could but that would involve having a custom license which pretty much comes with the same issues of writing a custom CLA. Probably even worse since licenses have to be legally accepted in a court of law to be enforced and you'd probably need a good lawyer in the first place to write that clause

All of the above is not really a CLA issue. A CLA allows you to re-license your code but you still need to respect the foss nature of it. There are no "free" license that outright disallow commercial usage but strong copyleft licenses like the gnu ones usually do the job.

CLA or not, even if you wrote 100% of Hyprland code with zero external contribution, there is no real way to legally enforce no commercial usage unless you use a proprietary license or turn the project into "source available". Something like the (A)GPL license would stop most companies from profiting of Hyprland but up to a certain point. You couldn't for example be able stop someone from profiting from selling a computer with Hyprland installed.

Then again using a CLA and a strong license would make it impossible for companies to build proprietary stuff on top of Hyprland like sony does with freebsd

Hopefully I made myself clear since I'm writing this a 2am

**Comment by morr0ne** [comment]
> Would it not be good to add something along the lines of "any program which uses this source in all or in part, regardless of commercial status, must make all associated code of said program open source"

That's pretty much what copyleft licenses do. Again this is a License issue. A CLA simply allows the owner of the project to re license the project if needed

**Comment by vaxerski** [comment]
makes sense, thanks for the explanation. I'll make some changes tomorrow to shift to fla.

**Comment by Sivecano** [comment]
> On a side note - since I write 95% of hyprland's code, wouldn't it be possible for me to one day add a clause that all _my_ contributions from here on out can't be used by commercial entities...?

if the commercial entites haven't previously obtained a license to your code you can.
I license only really denotes terms of use and redistribution but doesn't transfer ownership.

**Comment by port19x** [comment]
Other contributors already noted that you'd cease to be open source.
They didn't provide sufficient reasoning and proof, so you may be unsure about it.
Since I know a good bunch about software licensing I'll help you out here.

## Violations

In particular you'd be in violation of freedom 0 of the [free software definition](https://www.gnu.org/philosophy/free-sw.en.html)
> The freedom to run the program as you wish, for any purpose (freedom 0)

You would also be in violation of criteria 1 & 6 of the [open source definition](https://opensource.org/osd)
> (1) Free Redistribution
The license shall not restrict any party from selling or giving away the software as a component of an aggregate software distribution containing programs from several different sources. The license shall not require a royalty or other fee for such sale.
> (6) No Discrimination Against Fields of Endeavor
The license must not restrict anyone from making use of the program in a specific field of endeavor. For example, it may not restrict the program from being used in a business, or from being used for genetic research.

## Relicensing checklist

- [ ] Choose new license
- [ ] Add CLA or sth equivalent
- [ ] Get express permission of all prior contributors¹
- [ ] Relicense
- [ ] (optional) Announce and propagate to package authors

## Source-Available Licenses

- The BSL (Business Source License) is the one Hashicorp chose when un-opensourcing their products.²
- The Creative Commons Licenses have a NC (Non Commercial) modifier. This license is not good for code, but allows you to close down your docs and assets earlier.
- The [ethical source](https://ethicalsource.dev/) movement is another entity fighting against open source freedom. Perhaps they have a license that suits your needs.

## Be warned

Hyprland would be the first project I see getting good feedback for such a license change.
If you choose to really go that route, at least appease some of the offended by not claiming you're remaining open source.
This license change would impact distribution of the software in some linux distributions.
FSF endorsed distros are obvious candidates, but you'll also ruffle some feathers with debian and fedora.

<hr>

¹this has prevented some emacs packages, notably magit, from reassigning copyright to the FSF and becoming preinstalled

²https://news.ycombinator.com/item?id=37081306

**Comment by port19x** [comment]
:skull: 235 contributors. yeah, good luck. have fun getting sued if one of those is a salty neckbeard :skull:

**Comment by morr0ne** [comment]
@port19x yours are valid concerns but the FLA is designed specifically to avoid all these issues. You can read more about it in my comments above. It does address all the points and ensures Hyprland remains Free as in Freedom

**Comment by morr0ne** [comment]
It is also possible to properly relicense projects even without a CLA and it has been successfully done before.

https://github.com/grovesNL/glow/issues/212 is a great example where I had to give explicit permission to relicense my contributions

**Comment by port19x** [comment]
Thanks for pointing to these resources, you seem to know a fair bit about the CLA process.
In that case ignore my checklist.

Are we in agreement that "fully foss for individuals and non-profits" is, to the extent that it excludes for-profit business, a contradictory statement?
It might seem pedantic, but it's important that vaxry is fully aware that this makes hyprland source available and *not* free or open source software.

**Comment by port19x** [comment]
A one time license change like presented in your example looks great to me.
Better than CLA imho.

**Comment by mrfluffy-dev** [comment]
I will try really hard to articulate this in the correct wording. I just want to say I don't care what license you use for your project. It can be proprietary for all I care. But having contributors sign rights away leaving them with no future input with how there code is used is a bit iffy. a once off licensing change would be way better. it would be well understood coz all the terms would be laid down to the community and all passed contributors. And it wont be a Linus "Trust me bro" agreement.

**Comment by morr0ne** [comment]
@mrfluffy-dev I suggest you read the entire conversation carefully. As discussed above using the FLA protects contributors rights. It's not a "trust me bro" agreement but a proper legal document that makes legally enforceable guarantees

**Comment by morr0ne** [comment]
I would also like to clarify that contributing to a FOSS project does make any guarantee on how your code will be used. If anything the FLA gives contributors more protection by ensuring the project will always remain FOSS

**Comment by mrfluffy-dev** [comment]
I did read and I am fully ok with that. I do not intend to make anything you say invalid I intend to raise concerns. even a fully legal binding can be bent. look at Red hat for example. I just want to state that I think it would be better to allow only a once off license change. And I think that having more people talk about this the better. this is a change that holds the future of the project in its hands and it should not only be discussed buy 4 people. and not done over the course of a weekend. it is a hard challenge.

**Comment by ItsDrike** [comment]
For Hyprland, I'd say a complex CLA like the FLA isn't really that necessary. Since Hyprland is currently licensed under BSD 3 clause, which allows sublicensing already, you (or anyone else) can easily just go straight to some other, even to a closed source license.

If you do want to protect yourself against for-profit companies, you could immediately relicense to such a license, without any explicit approval from the contributors. That said, all of the code that is available now, while Hyprland is still under BSD 3 clause was licensed in perpetuity, which means anyone will be able to use it.

Only the newly added code, which would no longer be subject to this original license would be protected though. Also, depending on the new license, you might not even need a CLA once relicensed. The main purpose of CLAs is to allow owners/maintainers to gain the right to sublicene from the contributors through some other means than just the project license. This is done so that you, as the maintainer of the project, will still be able to relicense the project if you needed to, even if the license didn't allow it. But all of the existing contributions, including your own code, is currently perpetually licensed under BSD 3 clause, and will remain available under that license for anyone.

For that reason, introducing a CLA without also immediately changing the project's license is pretty much pointless at this point, your license allows you to relicense already. You will only need a CLA once you actually relicense, to keep the option to relicense in the future. So unless you already have a new license in mind, I'd say there's just no need for this.

**Comment by morr0ne** [comment]
> I did read and I am fully ok with that. I do not intend to make anything you say invalid I intend to raise concerns. even a fully legal binding can be bent. look at Red hat for example. I just want to state that I think it would be better to allow only a once off license change. 

Concerns are valid but I can't agree with your sentiment. Sure a legal document can be bent but if there isn't one at all that's even worse. Again the FLA does not sign away your rights like a normal CLA so stuff like the Red Hat situation should not happen.

> And I think that having more people talk about this the better. this is a change that holds the future of the project in its hands and it should not only be discussed buy 4 people. and not done over the course of a weekend. it is a hard challenge. 

In the end it is up to vaxyr, as he said before he pretty much wrote 95% of Hyprland code, this is essentially his project and he can do whatever we want, we can provide inputs but it's ultimately up to him

Besides all that, the project is already under the BSD-3-Clause license, it can already be easily abused by other corporations or vaxyr himself if he really wanted to. Most of your concerns have more to do with the specific license used. If you read the FLA in full you would easily realize that it protects contributors first and foremost.

**Comment by mrfluffy-dev** [comment]
The easy license change of BSD is what he wants to change. although you can only do this with a license that is compatible with BSD-3 and as far as I know non of the GPL licenses are for example so you can't use the AGPL (gpl for non commercial use) correct me if I am wrong.

**Comment by mrfluffy-dev** [comment]
@morr0ne Once again I did not state he can not do what he wants it is up to him and I did not state that your proposal is invalid. I stated he should be careful and keep the community in mind.

**Comment by ItsDrike** [comment]
> The easy license change of BSD is what he wants to change. although you can only do this with a license that is compatible with BSD-3 and as far as I know non of the GPL licenses are for example so you can't use the AGPL (gpl for non commercial use) correct me if I am wrong.

A CLA won't change anything about that, only a new license could do that, and only for the new code written from that point onwards. The only code will remain available under BSD 3-Clause.

BSD 3-Clause is a permissive license, it allows you to use the original (or modified) code in almost any way you like, including allowing you to sublicense (i.e. to change a license), that means you can even make the code proprietary. The only requirement it has is that you keep the copyright notice and original conditions mentioned somewhere in the project, but you don't have to follow those conditions, so you can relicense.

Yes, BSD 3-Clause isn't compatible with GPL, but that only matters when going from GPL -> BSD, not the other way around. That's why it can make sense to have a CLA with GPL (or other non-permissive) licensed projects, but it's just not needed with permissive ones.

**Comment by ItsDrike** [comment]
I will also point out that if you do add a CLA/DCO, you need to treat them as proper legal documents, otherwise there's really no point in having them at all. That means they will need to be legally signed by both you and the contributor. There's a fair few issues with that though:

- All contributors will need to be of legal age (usually 18+), otherwise, their legal guardians will need to sign it for them
- Contributors will need to sign with their real names, not github aliases, and they will likely need to also give an address or other identifying detail. Many could consider giving these information away a deal breaker when it comes to contributing.
- There will likely be a lot of hard to understand and long legal gibberish, which can scare away a lot of people.

**Comment by Aqa-Ib** [comment]
Yeah, ItsDrike got to the point. How is a CLA useful in a BSD licensed project ?

**Comment by mrfluffy-dev** [comment]
Then my proposal is if a licence change is inevitable although vax can decide without a care in the world. I think the correct choice would be a voting system with vaxry having the most say and contributors the second most with the community the least say like a point system. With the vote being between a few licenses or no license change or something like that. This keeps the community happy and the contributors. 


This is in no way a final idea obviously just a proposal of what could be done.

**Comment by Aqa-Ib** [comment]
Yeah, at this point it would be nice if vaxry tells us what the new license of the project will be.

**Comment by mrfluffy-dev** [comment]
Its only been a few hours. Mans still got a life and responsibilities other than this let him cook.

**Comment by Aqa-Ib** [comment]
I do not want to hurry things up. What I mean is: a CLA seems useless in a BSD licensed project, so he must have thought of changing the project's license before applying the CLA.

**Comment by vaxerski** [comment]
> Mans still got a life and responsibilities other than this let him cook.

Na sorry I was just sleeping. 

The entire goal of this, in my head, is to allow dual licensing in the case of me feeling like I am abused by some for-profit entities.

For example, leave BSD for private entities, but impose a different, restrictive license for corporate ones.

From what I can gather from this discussion, it seems to me like a bunch of you agree that BSD allows me to do that already...?

Tho I am not a lawyer, and maybe I should talk to one.

On a SN, I think it's not getting permission from all contributors to relicense to an incompatible license, but getting permission from all contributors whose code is still in the codebase. Those, whose code has been replaced, are technically not even being relicensed.

Let me know your thoughts.

**Comment by mrfluffy-dev** [comment]
Personal thoughts are do as you see fit it is your project. But for the good of your amd the projects reputation I think its best to involve the community in the decision. And talking to a lawyer might also be best if at all possible. Being informed with a decision like this is probably the most important part of it.

**Comment by ItsDrike** [comment]
> For example, leave BSD for private entities, but impose a different, restrictive license for corporate ones.

Due to the nature of BSD being permissive, you can't do that, only a copy-left (or a proprietary) license can reasonably achieve something like this, since BSD on it's own allows anyone to sublicense, including those for-profit companies.

> From what I can gather from this discussion, it seems to me like a bunch of you agree that BSD allows me to do that already...?

Yes BSD allows you or anyone to immediately change the license to some other license. Yes, you can also move to a dual license set up from it. You don't need anyone to sign a CLA to achieve that, but you will probably need a CLA once you move to this set up, since the entire point of that new license would then be to not allow such relicensing.

You can never dual license with a fully permissive OSI approved license in the way you describe. You could however move to GPL with a CLA, and because of the CLA, you will then have the option to use the contributions of others and relicense the code under something else. Some projects do this and have a version for commercial use, that companies can pay to get access to.

> 
> Tho I am not a lawyer, and maybe I should talk to one.

Indeed, that is always a good idea with these decisions.


> On a SN, I think it's not getting permission from all contributors to relicense to an incompatible license, but getting permission from all contributors whose code is still in the codebase. Those, whose code has been replaced, are technically not even being relicensed.

With the current BSD license, you don't need any kind of permission, the contributors gave you the right to relicense via that license.

If you did move to say GPL, you would however need a permission from everyone who owns any part of the code that you're attempting to relicense. That's where a CLA can be useful, since without it, the GPL license itself only allows you to use the code under it so long so all code that depends on it is also under GPL, and that's why you'd need a permission with it.

But note that the original code as it is now is fully BSD, and you (or anyone) will have the option to use it like that forever. Also, even after a relicense to say GPL, you could relicense afterwards easily, if only the future contributors signed the CLA, since the old ones gave you the code under BSD, which again allows relicensing.

---

That all said, I'm opposed to introducing a CLA here, because of the issues with signing it in a proper legal way, which would require a real life name and details. I'm sure I'm not alone in being reluctant in giving these out just to be able to contribute to some project.

**Comment by port19x** [comment]
Remaining foss seems more valuable to me than preventing some theoretical corporate exploitation.
GPL might make sense.
A relicense is probably neither worth the time or gain in the first place

**Comment by i509VCB** [comment]
> * Contributors will need to sign with their real names, not github aliases, and they will likely need to also give an address or other identifying detail. Many could consider giving these information away a deal breaker when it comes to contributing.

The requirement of a real name has arguably been yeeted since Linux allows contributions from contributors with pseudonyms such as Asahi Lina.

**Comment by ItsDrike** [comment]
> > * Contributors will need to sign with their real names, not github aliases, and they will likely need to also give an address or other identifying detail. Many could consider giving these information away a deal breaker when it comes to contributing.
> 
> The requirement of a real name has arguably been yeeted since Linux allows contributions from contributors with pseudonyms such as Asahi Lina.

It is **very** rare to see a CLA that works with pseudonyms, Linux Kernel only has a DCO, even there it's a bit controversial.

CLA is a legally binding document, unlike a software license, where you as the author automatically give the rights to anyone as per that license, a CLA is a contract between some entity (the maintainer), and you, specifying conditions that both sides have to adhere to.

To enforce such a document in a court of law, you need to be able to know who this contract was actually signed with, and who's copyright is even being transfered. Without this information a CLA is pretty much useless and likely won't be enforceable.

As a reference, see this issue of the whatwg project, that wasn't able to be solved by their steering council: https://github.com/whatwg/sg/issues/93

Similarly, see this stack exchange question: https://opensource.stackexchange.com/questions/2847/contributor-license-agreements-and-anonymous-pseudonymous-contributors where the answer also mentions the need to know the real names.

At most, you could have a clause in the CLA, where the maintainer agrees to not reveal your real life name anywhere except for when taking actual legal actions based on that CLA, with perhaps something like a good faith agreement.

That might satisfy some contributors, but probably not all, and you still face the issue of the contributor having to be of legal age, and getting them to sign a legally binding document written in relatively complex legalese language.

**Comment by Thaodan** [comment]
I can't add so much to the discussion  as most things have been already said but there's another thing:
CLA's increase the burden of contribution to fill out the paper work for each new contributor.

I don't see any danger of "abuse" from commercial users in this project with copyleft in place.

**Comment by ArrayBolt3** [comment]
All of this seems like an awful lot of complexity for a simple goal - make something free that people can use for whatever, but keep a commercial entity from coming along and "stealing" it. Trying to procure the rights to relicensing is possible, but it makes it harder for people to contribute, and sure, maybe *you* won't pull a MongoDB on everyone, but there's no guarantee you won't end up with the rights to your code transferred to someone else (for instance, when you die), and whoever gets those rights may very well do that. From what I gather it sounds like the FLA is supposed to help avoid that maybe? But still, you're trying to swat a fly with a sledgehammer here IMO.

You say that the GPL is too restrictive. But... why? What use case can you see being hampered by the GPL that you want to allow? For the average user who just wants to use the code, the GPL is entirely unrestrictive - people can use your project for whatever. For the power user who wants to mod it, they're free to do that. For the contributor who wants to help out, they can do that too. For the person who wants to fork it and make something totally different with the shared code, that's doable too. For the company that wants to commercialize it, they're perfectly able to do that if they're willing to be a good open-source citizen. Things get slightly sticky when trying to incorporate code into other pre-existing projects, but if you use a GPL version 2 **or later** clause, most of that stickiness vanishes. The only people who are really put out by the GPL are those who want to mod the code *and* distribute binaries *and* keep the code to themselves. For those people, the GPL is a huge thorn in their side, and that's exactly what it's intended to be. (Things can also become painful for people who want to combine code from projects with incompatible GPL licensing terms, but again, use GPLv2+ and those issues become way less severe.)

The only good reason to use a permissive license is to specifically let businesses take things and make them proprietary. If you want that sort of thing to happen (which it sounds like you really don't), go permissive. If not, go GPL. What you're trying to do is what the GPL was made for, and it's still the best tool for the job IMO.

(I'm an avid open-source software dev, an Ubuntu MOTU, and do a lot of license compliance auditing as part of my packaging work in Ubuntu and Debian. I am **not** a lawyer.)

**Comment by LechintanTudor** [comment]
If you find GPL too restrictive, you should consider using [MPL](https://www.mozilla.org/en-US/MPL/2.0/).

**Comment by owittnan** [comment]
I can't understand how you simultaneously want to restrict corporations use of Hypr code but call the GPL "too restrictive". It seems to me like the GPL does exactly what you want.

What is "too restrictive" about the GPL that makes it unsuitable for use here?

**Comment by SViN24** [comment]
You might want to consider LGPL/MPL if you find GPL too restrictive.

Also you might want to clarify what properties future versions of said license will have in the CLA. 
In addition to that You might want to clarify clause 1 of the CLA.

>  Ⅰ) This contribution is made entirely by myself, or in a way that gives me full rights to the contribution.

It is not entirely legally clear who "myself" and "me" refers to.
For example you'll see the following line in some microsoft EULA things.

> “We,” “our,” and “us” includes Microsoft, the device manufacturer, and software ...

Before pushing something like this you might want to talk to a lawyer.

**Comment by morr0ne** [comment]
@vaxerski would you consider closing/locking this issue? The conversation has already happened with the reached conclusion that talking to a lawyer is needed first. People are clearly coming here from reddit and commenting without reading the rest of the thread and missing the entire context of the discord server. It doesn't seem like anything meaningful is being added to the conversation that hasn't already been said before, at this point it's just spam

**Comment by vaxerski** [comment]
Locking is a bit much, but yeah I'll close it, I'll talk to a lawyer prolly this summer. 

Thanks for all the input :)
